partner_contact_in_several_companies/models/ir_actions.py

```diff
@@ -1,24 +1,23 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 from odoo import models
+from odoo.tools.safe_eval import safe_eval
 
 
 class IRActionsWindow(models.Model):
     _inherit = "ir.actions.act_window"
 
     def read(self, fields=None, context=None, load="_classic_read"):
         actions = super(IRActionsWindow, self).read(fields=fields, load=load)
-        for action in actions:
-            if action.get("res_model", "") == "res.partner":
-                # By default, only show standalone contact
-                action_context = action.get("context", "{}")
-                if "search_show_all_positions" not in action_context:
-                    action["context"] = action_context.replace(
-                        "{",
-                        (
-                            "{'search_show_all_positions': "
-                            "{'is_set': True, 'set_value': False},"
-                        ),
-                        1,
-                    )
+        for action in filter(
+            lambda rec: rec.get("res_model") == "res.partner", actions
+        ):
+            # By default, only show standalone contact
+            action_context = safe_eval(action.get("context", "{}"))
+            if "search_show_all_positions" not in action_context:
```

Comment:
You could use this for check key in dict: `action_context.get("search_show_all_positions")`